src/sound/s_sound.cpp

```diff
@@ -1235,6 +1235,10 @@ static FSoundChan *S_StartSound(AActor *actor, const sector_t *sec, const FPolyO
 		default:										break;
 		}
 	}
+
+	if (spitch > 0.0)
+		S_SetPitch(chan, spitch);
```

Comment:
This seems a bit weird to apply the specified `spitch` after the sound has already started (depending on the timing, it may or may not result in the sound starting at the normal pitch before changing to the desired one). I understand why it's done (the `StartSound[3D]` method will completely ignore the starting pitch if `snd_pitched` is false), though I wonder if it may be worth dealing with that along with this PR. It seems to me that `snd_pitched` could be handled out here in `s_sound.cpp` so the sound renderer can always use the starting pitch as given.

Other than that, though, I don't notice anything obviously wrong with the PR.

Reply:
No, actually, the sound is started before the pitch is applied. I did this for convenience and fewer places to modify this. In particular, places with S_Sound calls directly so this maintains compatibilty for ACS too.

Furthermore, ZDoom does it even earlier, take a look at the raw file. You'll see the pitch is set MUCH earlier.